### PR TITLE
Improve Error List message readability

### LINKED ISSUE #360: Improve readability of Sarif result message in Error List
Comments from Michael:
* We're truncating messages in the error list so that they're too short. Let's increase that.  
* We should try to force the error list forward (make it visible and active) if anything was sent to the UI.

### PR DESCRIPTION
**Pull request by yongyan-gh**
Fixes #360 

* Increase the max length of message display in error list table entry. from 120 to 165.
* Do not split message into shorter sentences and show first sentence in table entry. try to show full message as possible in the table entry, only separate message into 2 parts if full message length is longer than max length.

### CODE REVIEW

File: src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs [hunk 1]
```diff
@@ -673,29 +673,28 @@ internal IEnumerable<string> GetCodeSnippets()
             return this.Locations.Select(location => location.ExtractSnippet(regionsCache));
         }
 
-        internal (string first, string second) SplitMessageText(string fullText, int maxLength = 120)
+        internal (string first, string second) SplitMessageText(string fullText, int maxLength = 165)
         {
-            string shortText = ExtensionMethods.GetFirstSentence(fullText);
+            // remove line breakers
+            fullText = fullText.Replace(Environment.NewLine, " ").Replace("\r", " ").Replace("\n", " ");
```
Comment (eddynaka) [inline src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs:679]: >fullText = fullText.Replace(Environment.NewLine, " ").Replace("\r", " ").Replace("\n", " "); [](start = 12, length = 92)

I'm not sure about this.

File: src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifErrorListItemTests.cs [hunk 2]
```diff
@@ -380,6 +383,44 @@ public void SarifErrorListItem_ResultMessageFormat_NoTrailingPeriod()
...
+            };
+
+            SarifErrorListItem item = MakeErrorListItem(result);
+            int breakPoistion = 165;
+            item.ShortMessage.Length.Should().Be(breakPoistion + 2);
+            item.Message.Length.Should().Be(200 - breakPoistion);
+        }
+
+        [Fact]
+        public void SarifErrorListItem_ResultMessageFormat_LongTextWitBreak()
```
Comment (eddynaka) [inline src/Sarif.Viewer.VisualStudio.UnitTests/ErrorList/SarifErrorListItemTests.cs:406]: >SarifErrorListItem_ResultMessageFormat_LongTextWitBreak [](start = 20, length = 55)

can you create a test using environment new line, \r and \n? that would check the method we updated

File: src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs (hunk 1 above)
Comment (yongyan-gh) [reply src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs:679]: Environment.NewLine (\r\n) is windows sty, \n and \r are unix/mac style line breakers. I see sarif-sdk code handles all of 3 so I did the similar here.

---
In reply to: [573064240](https://github.com/microsoft/sarif-visualstudio-extension/pull/371#discussion_r573064240) [](ancestors = 573064240)

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**